File: ios/ShortDrama/Sources/Domain/UseCases/SearchDramasUseCase.swift
import Foundation

/// Use case for searching dramas by keyword.
struct SearchDramasUseCase: Sendable {

    private let repository: DramaRepositoryProtocol

    init(repository: DramaRepositoryProtocol) {
        self.repository = repository
    }

    func execute(query: String, page: Int, pageSize: Int) async throws -> [Drama] {
        try await repository.searchDramas(query: query, page: page, pageSize: pageSize)
    }
}

File: android/app/src/main/java/com/djs66256/short_drama/domain/usecase/SearchDramasUseCase.kt
package com.djs66256.short_drama.domain.usecase

import com.djs66256.short_drama.core.network.ApiResult
import com.djs66256.short_drama.domain.model.Drama
import com.djs66256.short_drama.domain.repository.SearchRepository
import javax.inject.Inject

class SearchDramasUseCase @Inject constructor(
    private val searchRepository: SearchRepository,
) {
    suspend operator fun invoke(
        query: String,
        page: Int = 1,
        pageSize: Int = 10,
    ): ApiResult<List<Drama>> = searchRepository.searchDramas(query, page, pageSize)
}
